### DISCUSSION TITLE
Hypervisior && Programming language

### DISCUSSION THREAD

**Discussion by undolog** [manoelcampos/cloudsim-plus#486] [Q&A] [answered]
Hello developers, i have 2 questions about cloudsimplus.

1)  What kind of architecture is the virtual machine simulated by this platform based on, or universal？
Different kinds of hypervisor have different requirements in terms of power consumption overhead performance.

2) Does this SDK have a python language version? 
I did not find an obvious hint on the official website.

Thanks for answering.

**Comment by manoelcampos** [comment, marked as answer]
> 1. What kind of architecture is the virtual machine simulated by this platform based on, or universal？
>    Different kinds of hypervisor have different requirements in terms of power consumption overhead performance.

There is no defined architecture of VMs or hypervisors. The power and computing requirements is totally up to you.
You define how VMs consume resources such as RAM or CPU (by means of running Cloudlets) and how physical machines consume power (based on VMs CPU utilization). Check the power features on the README "2. Main Exclusive Features" section.

> 2. Does this SDK have a python language version?
>    I did not find an obvious hint on the official website.

Yes, it does. Just check the "Related Projects" section on README.

**Reply by undolog** [reply]
Sincerely thank you for your reply. I will refer to the documentation for configuration experiments.
